src/Form/LegacyConsumer/resources/js/conditinal-fields.js

```diff
@@ -0,0 +1,222 @@
+window.addEventListener('load', async () => {
+	const state = {};
+
+	/**
+	 * Get list of watched fields.
+	 * @unreleased
+	 *
+	 * @return object
+	 */
+	function getWatchedElementNames(donationForm) {
+		const fields = {};
+
+		donationForm.querySelectorAll('[data-field-visibility-conditions]').forEach(function (inputField) {
+			const visibilityConditions = JSON.parse(inputField.getAttribute('data-field-visibility-conditions'));
+			const visibilityCondition = visibilityConditions[0]; // Currently we support only one visibility condition.
+			const {field} = visibilityCondition;
+
+			fields[field] = {
+				...fields[field],
+				[inputField.name]: visibilityConditions
+			}
+		})
+
+		return fields;
+	}
+
+	/**
+	 * @unreleased
+	 *
+	 * @param operator
+	 * @param firstData
+	 * @param secondData
+	 *
+	 * @return boolean
+	 */
+	function compareWithOperator( operator, firstData, secondData ){
+		return {
+			'=': firstData === secondData,
+			'!=': firstData != secondData,
```

Comment:
Can we use `!==` operator instead of `!=` because all the time form field value will be`string`, right?
I took these operators from ` BasicCondition::OPERATORS`.

Reply:
@JasonTheAdams need your suggestion on this.

Reply:
@ravinderk do you expect any other types besides string?

Reply:
I do not think we have any plan (now) to support multiple data types in comparision.